tests/onnxruntime/test_modeling.py

```diff
@@ -2009,12 +2019,13 @@ def test_merge_from_transformers_and_save(self, model_arch):
         model = ORTModelForCausalLM.from_pretrained(model_id, export=True, use_merged=True)
         with tempfile.TemporaryDirectory() as tmpdir:
             model.save_pretrained(tmpdir)
-            save_path = os.path.join(tmpdir, ONNX_DECODER_MERGED_NAME)
-            self.assertTrue(has_onnx_input(save_path, "use_cache_branch"))
+            save_path = os.path.join(tmpdir, ONNX_WEIGHTS_NAME)
+            self.assertFalse(has_onnx_input(save_path, "use_cache_branch"))
 
             folder_contents = os.listdir(tmpdir)
             self.assertTrue(ONNX_DECODER_NAME not in folder_contents)
             self.assertTrue(ONNX_DECODER_WITH_PAST_NAME not in folder_contents)
+            self.assertTrue(ONNX_DECODER_MERGED_NAME not in folder_contents)
```

Comment:
I'm not sure to understand. `use_merged=True`, `save_pretrained` and the merged decoder not in the saved folder?

Reply:
`use_merged=True` was deprecated for export via `ORTModel`, but still available via the CLI (with `--legacy`), here we just verify that the model was exported as expected and not merged, which is not super useful, might make sense to remove it!

Reply:
If i may ask this question:
> use_merged=True was deprecated for export via ORTModel, but still available via the CLI (with --legacy)

Does this mean that merging model has been proven to be non effective and not supported in a near future (as the legacy word seems to indicate ?)

Thanks,

Reply:
for decoder only models, the decoder will now be exported only once by default, previously we had to export the model twice (with and without kv cache) before we could create a "merged" model which was quite memory intensive. do you have a case where you need to use a merged model / cannot use the now default export ?